server/src/daemon/prompt.rs

```diff
@@ -0,0 +1,139 @@
+// org.freedesktop.Secret.Prompt
+
+use std::{
+    collections::HashMap,
+    sync::{Arc, Mutex},
+};
+
+use oo7::portal::{Keyring, Secret};
+use serde::{Serialize, Serializer};
+use tokio;
+use zbus::{
+    fdo, interface,
+    message::Header,
+    proxy,
+    zvariant::{self, DeserializeDict, ObjectPath, OwnedObjectPath, SerializeDict, Type, Value},
+    Connection, SignalContext,
+};
+
+use super::{
+    prompter::{PrompterCallback, PrompterProxy, Properties},
+    secret_exchange::{get_secret, SecretExchange},
+    service_manager::ServiceManager,
+};
+use crate::{LOGIN_KEYRING, LOGIN_KEYRING_PATH, SECRET_PROMPT_PREFIX};
+
+#[derive(Default, DeserializeDict, Debug, Type, SerializeDict)]
+#[zvariant(signature = "dict")]
+pub struct PromptResult {
+    path: OwnedObjectPath,
+}
+
+#[derive(Clone, Debug, Default, zvariant::Type)]
+#[zvariant(signature = "o")]
+pub struct Prompt {
+    manager: Arc<Mutex<ServiceManager>>,
+    path: OwnedObjectPath,
+}
+
+#[interface(name = "org.freedesktop.Secret.Prompt")]
+impl Prompt {
+    pub async fn prompt(
+        &self,
+        window_id: &str,
+        #[zbus(object_server)] object_server: &zbus::ObjectServer,
+        #[zbus(connection)] connection: &zbus::Connection,
+        #[zbus(header)] header: Header<'_>,
+    ) -> fdo::Result<()> {
+        println!("prompt: {}", header.path().unwrap());
+
+        let post_fix = if header.path().unwrap().as_str().to_string().contains("/u") {
+            Some("u")
+        } else {
+            None
+        };
```

Comment:
This is horrible, just have two different kinds of Prompts.

Reply:
yeah :) implemented two Prompt constructors.